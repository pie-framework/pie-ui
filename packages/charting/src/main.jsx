import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { color, Collapsible } from '@pie-lib/render-ui';
import { Chart, chartTypes } from '@pie-lib/charting';
import isEqual from 'lodash/isEqual';

export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object,
    model: PropTypes.object.isRequired,
    onAnswersChange: PropTypes.func,
    categories: PropTypes.array,
  };

  static defaultProps = { classes: {} };

  constructor(props) {
    super(props);

    this.state = {
      categories: props.categories || props.model.data,
    };
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    const { model: { data: nextData = [] } = {} } = nextProps;
    const { model: { data = [] } = {} } = this.props;

    if (!isEqual(nextData, data)) {
      this.setState({ categories: nextData });
    }
  }

  changeData = (data) =>
    this.setState(
      {
        categories: data,
      },
      () => this.props.onAnswersChange(data)
    );

  render() {
    const { categories } = this.state;
    const { model, classes } = this.props;
    const {
      teacherInstructions,
      prompt,
      chartType,
      size,
      domain,
      range,
      title,
      editCategoryEnabled,
      addCategoryEnabled,
      categoryDefaultLabel,
      rationale,
      correctedAnswer,
    } = model;

    return (
      <div className={classes.mainContainer}>
        {teacherInstructions && (
          <Collapsible
            labels={{
              hidden: 'Show Teacher Instructions',
              visible: 'Hide Teacher Instructions',
            }}
          >
            <div dangerouslySetInnerHTML={{ __html: teacherInstructions }} />
          </Collapsible>
        )}
        <br />
        <div
          className={classes.prompt}
          dangerouslySetInnerHTML={{ __html: prompt }}
        />

        <br />

        <Chart
          chartType={chartType}
          size={size}
          domain={domain}
          range={range}
          charts={[
            chartTypes.Bar(),
            chartTypes.Histogram(),
            chartTypes.LineDot(),
            chartTypes.LineCross(),
            chartTypes.DotPlot(),
            chartTypes.LinePlot(),
          ]}
          data={correctedAnswer || categories}
          title={title}
          onDataChange={this.changeData}
          editCategoryEnabled={editCategoryEnabled}
          addCategoryEnabled={addCategoryEnabled}
          categoryDefaultLabel={categoryDefaultLabel}
        />

        <br />
        {rationale && (
          <Collapsible
            labels={{ hidden: 'Show Rationale', visible: 'Hide Rationale' }}
          >
            <div dangerouslySetInnerHTML={{ __html: rationale }} />
          </Collapsible>
        )}
      </div>
    );
  }
}

const styles = theme => ({
  mainContainer: {
    padding: theme.spacing.unit,
    color: color.text(),
    backgroundColor: color.background(),
  },
  prompt: {
    verticalAlign: 'middle',
  },
});

export default withStyles(styles)(Main);
