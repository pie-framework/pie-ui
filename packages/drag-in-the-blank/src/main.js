import React from 'react';
import PropTypes from 'prop-types';
import isEmpty from 'lodash/isEmpty';
import CorrectAnswerToggle from '@pie-lib/correct-answer-toggle';
import { DragInTheBlank } from '@pie-lib/mask-markup';
import { withDragContext } from '@pie-lib/drag';
import { Collapsible } from '@pie-lib/render-ui';

const DraggableDragInTheBlank = withDragContext(DragInTheBlank);

export class Main extends React.Component {
  static propTypes = {
    prompt: PropTypes.string,
    disabled: PropTypes.bool,
    markup: PropTypes.string,
    model: PropTypes.object,
    value: PropTypes.object,
    feedback: PropTypes.object,
    onChange: PropTypes.func
  };

  static defaultProps = {
    value: {}
  };

  state = {
    showCorrectAnswer: false
  };

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (isEmpty(nextProps.feedback)) {
      this.setState({ showCorrectAnswer: false });
    }
  }

  toggleShowCorrect = () => {
    this.setState({ showCorrectAnswer: !this.state.showCorrectAnswer });
  };

  render() {
    const { showCorrectAnswer } = this.state;
    const { model, onChange, value } = this.props;
    const { prompt, mode } = model;
    const modelWithValue = {
      ...model,
      value
    };

    return (
      <div>
        {
          model.teacherInstructions && (
            <Collapsible labels={{ hidden: 'Show Teacher Instructions', visible: 'Hide Teacher Instructions' }}>
              <div dangerouslySetInnerHTML={{ __html: model.teacherInstructions }}/>
            </Collapsible>
          )
        }
        <br />
        <CorrectAnswerToggle
          show={mode === 'evaluate'}
          toggled={showCorrectAnswer}
          onToggle={this.toggleShowCorrect}
        />
        {prompt && <div dangerouslySetInnerHTML={{ __html: prompt }}/>}
        <DraggableDragInTheBlank
          {...modelWithValue}
          onChange={onChange}
          showCorrectAnswer={showCorrectAnswer}
        />
      </div>
    );
  }
}

export default Main;
