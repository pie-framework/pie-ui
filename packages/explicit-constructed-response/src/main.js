import React from 'react';
import PropTypes from 'prop-types';
import isEmpty from 'lodash/isEmpty';
import CorrectAnswerToggle from '@pie-lib/correct-answer-toggle';
import { ConstructedResponse } from '@pie-lib/mask-markup';
import { Collapsible } from '@pie-lib/render-ui';

export class Main extends React.Component {
  static propTypes = {
    prompt: PropTypes.string,
    disabled: PropTypes.bool,
    markup: PropTypes.string,
    mode: PropTypes.string,
    teacherInstructions: PropTypes.string,
    value: PropTypes.object,
    feedback: PropTypes.object,
    onChange: PropTypes.func
  };

  static defaultProps = {
    value: {}
  };

  state = {
    showCorrectAnswer: false
  };

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (isEmpty(nextProps.feedback)) {
      this.setState({ showCorrectAnswer: false });
    }
  }

  toggleShowCorrect = () => {
    this.setState({ showCorrectAnswer: !this.state.showCorrectAnswer });
  };

  render() {
    const { showCorrectAnswer } = this.state;
    const { mode, prompt, teacherInstructions } = this.props;

    return (
      <div>
        {
          teacherInstructions && (
            <div style={{ margin: '16px 0' }}>
              <Collapsible
                labels={{ hidden: 'Show Teacher Instructions', visible: 'Hide Teacher Instructions' }}
              >
                <div dangerouslySetInnerHTML={{ __html: teacherInstructions }}/>
              </Collapsible>
            </div>
          )
        }
        <CorrectAnswerToggle
          show={mode === 'evaluate'}
          toggled={showCorrectAnswer}
          onToggle={this.toggleShowCorrect}
        />
        {prompt && <div dangerouslySetInnerHTML={{ __html: prompt }}/>}
        <ConstructedResponse
          {...this.props}
          showCorrectAnswer={showCorrectAnswer}
        />
      </div>
    );
  }
}

export default Main;
