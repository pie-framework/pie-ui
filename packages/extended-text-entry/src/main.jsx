import React from 'react';
import EditableHTML from '@pie-lib/editable-html';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import debug from 'debug';
import { Feedback, Collapsible } from '@pie-lib/render-ui';

const log = debug('@pie-ui:extended-text-entry');

const style = theme => ({
  main: {
    margin: '20px'
  },
  prompt: {
    width: '100%',
    marginBottom: theme.spacing.unit * 2
  }
});

class Main extends React.Component {
  static propTypes = {
    onChange: PropTypes.func.isRequired,
    model: PropTypes.object,
    classes: PropTypes.object.isRequired,
    session: PropTypes.shape({
      value: PropTypes.string
    }).isRequired
  };

  render() {
    const { model, onChange, classes, session } = this.props;
    const { width, height, disabled, feedback, teacherInstructions } = model;
    const { value } = session;
    log('[render] disabled? ', disabled);
    return (
      <div className={classes.main}>
        {
          teacherInstructions && (
            <Collapsible
              labels={{ hidden: 'Show Teacher Instructions', visible: 'Hide Teacher Instructions' }}
              className={classes.collapsible}
            >
              <div dangerouslySetInnerHTML={{ __html: teacherInstructions }}/>
            </Collapsible>
          )
        }
        <br />
        {model.prompt && (
          <Typography
            className={classes.prompt}
            dangerouslySetInnerHTML={{ __html: model.prompt }}
          />
        )}
        <EditableHTML
          onChange={onChange}
          markup={value || ''}
          width={width && width.toString()}
          height={height && height.toString()}
          disabled={disabled}
          highlightShape={true}
        />
        <br />
        {feedback && <Feedback correctness="correct" feedback={feedback} />}
      </div>
    );
  }
}

export default withStyles(style)(Main);
