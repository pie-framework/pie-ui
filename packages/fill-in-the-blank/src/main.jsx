import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import { color } from '@pie-lib/render-ui';
import Content from './content';
import Choices from './choices';
import { buildContent, addChoice, removeChoice } from './builder';
import CorrectAnswerToggle from '@pie-lib/correct-answer-toggle';

import { withDragContext, uid } from '@pie-lib/drag';
import debug from 'debug';

const log = debug('@pie-ui:fill-in-the-blank:main');

export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    model: PropTypes.object.isRequired,
    session: PropTypes.object.isRequired,
    uid: PropTypes.string,
    onAnswersChange: PropTypes.func.isRequired
  };
  static defaultProps = {};

  constructor(props) {
    super(props);
    this.uid = props.uid || uid.generateId();
    this.state = {
      showCorrect: false
    };
  }

  dropChoice = (blankId, choice) => {
    const { model, session, onAnswersChange } = this.props;
    log('dropChoice: blank:', blankId, 'choice: ', choice);
    session.answers = session.answers || [];

    session.answers = addChoice(
      choice.id,
      blankId,
      session.answers,
      model.choices,
      choice.blankId
    );

    onAnswersChange(session.answers);
  };

  removeChoice = (blankId, choice) => {
    const { session, onAnswersChange } = this.props;
    session.answers = session.answers || [];
    session.answers = removeChoice(choice.id, blankId, session.answers);
    onAnswersChange(session.answers);
  };

  toggleShowCorrect = () => {
    this.setState({ showCorrect: !this.state.showCorrect });
  };

  render() {
    const { classes, className, model, session } = this.props;
    const { showCorrect } = this.state;
    const { content, choices } = buildContent(
      model,
      showCorrect ? model.correctResponse : session.answers || []
    );

    return (
      <uid.Provider value={this.uid}>
        <div className={classes.mainContainer}>
          <CorrectAnswerToggle
            show={model.correctResponse !== undefined}
            toggled={showCorrect}
            onToggle={this.toggleShowCorrect}
            className={classes.toggle}
          />
          <div className={classNames(classes.main, className)}>
            <Content
              content={content}
              onDropChoice={this.dropChoice}
              onRemoveChoice={this.removeChoice}
              disabled={model.disabled}
            />
            <Choices
              choices={choices}
              label={model.choiceLabel}
              disabled={model.disabled}
            />
          </div>
        </div>
      </uid.Provider>
    );
  }
}

const styles = theme => ({
  mainContainer: {
    padding: theme.spacing.unit,
    color: color.text(),
    backgroundColor: color.background()
  },
  main: {},
  toggle: {
    paddingBottom: theme.spacing.unit
  }
});

export default withDragContext(withStyles(styles)(Main));
