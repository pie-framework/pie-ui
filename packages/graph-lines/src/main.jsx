import React from 'react';
import { GraphLines, lineUtils as utils } from '@pie-lib/charting';
import PropTypes from 'prop-types';
import Controls from './controls';
import CorrectAnswerToggle from '@pie-lib/correct-answer-toggle';
import { Feedback } from '@pie-lib/render-ui';
import GraphLineControls from './GraphLineControls';
import { withStyles } from '@material-ui/core/styles';

export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object,
    session: PropTypes.shape({
      lines: PropTypes.arrayOf(
        PropTypes.shape({
          from: PropTypes.shape({
            x: PropTypes.number.isRequired,
            y: PropTypes.number.isRequired
          }),
          to: PropTypes.shape({
            x: PropTypes.number.isRequired,
            y: PropTypes.number.isRequired
          })
        })
      )
    }),
    onSessionChange: PropTypes.func,
    model: PropTypes.object.isRequired
  };

  constructor(props) {
    super(props);

    const lines = props.session.lines || [];

    this.state = {
      session: { ...props.session, lines },
      selection: [],
      useSessionLines: false,
      showCorrect: false
    };
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    const lines = nextProps.session.lines || [];
    let shouldUseSessionLines = true;

    // if we do not have lines in the session OR
    // if we previously had lines in the session and they were cleared out
    // it means the model has changed and we need to stop using the lines in the session
    if (
      !this.props.session.lines ||
      (this.props.session.lines &&
        this.props.session.lines.length &&
        this.state.useSessionLines &&
        !lines.length)
    ) {
      shouldUseSessionLines = false;
    }

    const session = { ...nextProps.session, lines };

    this.setState({ session, useSessionLines: shouldUseSessionLines });
  }

  setUseSessionLines = lines => {
    this.setState(
      state => ({
        session: { ...state.session, lines },
        useSessionLines: true
      }),
      this.callOnSessionChange
    );
  };

  callOnSessionChange = () => {
    const { onSessionChange } = this.props;

    if (onSessionChange) {
      onSessionChange(this.state.session);
    }
  };

  deleteSelection = () => {
    const lines = utils.removeLines(
      this.state.session.lines,
      this.state.selection
    );

    const session = { ...this.state.session, lines };
    this.setState({ session, selection: [] }, this.callOnSessionChange);
  };

  buildLines = () => {
    const { model } = this.props;
    const { session, showCorrect, selection, useSessionLines } = this.state;

    if (model.correctResponse) {
      if (showCorrect) {
        return model.correctResponse;
      } else {
        return session.lines.map(line => ({
          ...line,
          selected: utils.hasLine(selection, line),
          correct: utils.hasLine(model.correctResponse, line)
        }));
      }
    } else if (useSessionLines) {
      return session.lines;
    } else {
      const lines = [];

      model.graph.lines.forEach(line => {
        const lineExpression = utils.expressionFromDescriptor(line.initialView);
        const points = utils.pointsFromExpression(lineExpression);

        lines.push({ ...line, ...points });
      });

      return lines;
    }
  };

  toggleShowCorrect = show => {
    this.setState({ showCorrect: show });
  };

  onLineChange = (old, newLine) => {
    const { session, selection, useSessionLines } = this.state;
    const linesToUse = useSessionLines ? session.lines : this.buildLines();
    const newLines = utils.swapLine(linesToUse, old, newLine);
    const newSelection = utils.swapLine(selection, old, newLine);
    const newSession = { ...session, lines: newLines };

    this.setState(
      { session: newSession, selection: newSelection, useSessionLines: true },
      this.callOnSessionChange
    );
  };

  toggleSelectLine = line => {
    const { session, useSessionLines, selection } = this.state;
    let newLines;
    let newSelection;

    if (utils.hasLine(selection, line)) {
      newSelection = utils.removeLine(selection, line);
    } else {
      newSelection = selection.concat([{ ...line, selected: true }]);
    }

    const linesToUse = useSessionLines ? session.lines : this.buildLines();

    newLines = linesToUse.map(lineToUse => {
      if (utils.linesEqual(lineToUse, line)) {
        return { ...lineToUse, selected: !utils.hasLine(selection, line) };
      }

      return lineToUse;
    });

    const newSession = { ...session, lines: newLines };

    this.setState(
      { selection: newSelection, session: newSession, useSessionLines: true },
      this.callOnSessionChange
    );
  };

  onAddPoint = () => {
    const linesFromModel = this.buildLines();

    this.setUseSessionLines(linesFromModel);
  };

  onAddLine = line => {
    this.setState(
      state => ({
        session: { ...state.session, lines: [...state.session.lines, line] }
      }),
      this.callOnSessionChange
    );
  };

  updateLines = newLines => {
    this.setUseSessionLines(newLines);
  };

  onDeleteLine = line => {
    const { session, useSessionLines, selection } = this.state;
    const linesToUse = useSessionLines ? session.lines : this.buildLines();

    this.setState(
      state => ({
        session: {
          ...state.session,
          lines: utils.removeLine(linesToUse, line)
        },
        useSessionLines: true,
        selection: utils.removeLine(selection, line)
      }),
      this.callOnSessionChange
    );
  };

  render() {
    const { model, classes } = this.props;
    const { selection, showCorrect } = this.state;
    const lines = this.buildLines();

    return (
      <div className={classes.mainContainer}>
        <div className={classes.main}>
          {model.correctness && <div>Score: {model.correctness.score}</div>}
          <CorrectAnswerToggle
            className={classes.toggle}
            show={
              !model.exhibitOnly &&
              model.correctness &&
              model.correctness.correctness !== 'correct'
            }
            toggled={showCorrect}
            onToggle={this.toggleShowCorrect}
          />
          {!model.disabled &&
            !model.exhibitOnly && (
              <Controls
                iconOnly={false}
                disabled={!(selection && selection.length > 0)}
                onDeleteClick={this.deleteSelection}
              />
            )}
          <GraphLineControls
            lines={lines}
            model={model}
            onDeleteLine={this.onDeleteLine}
            updateLines={this.updateLines}
          />
          <GraphLines
            maxLines={model.multiple ? model.lines.length : 1}
            className={classes.graph}
            lines={lines}
            width={model.width}
            height={model.height}
            domain={model.domain}
            range={model.range}
            disabled={model.disabled || model.exhibitOnly || false}
            onAddPoint={this.onAddPoint}
            onAddLine={this.onAddLine}
            onLineChange={this.onLineChange}
            onLineClick={this.toggleSelectLine}
          />
        </div>
        {model.feedback && (
          <Feedback
            correctness={model.correctness.correctness}
            feedback={model.feedback}
            width={model.width - 20}
          />
        )}
      </div>
    );
  }
}

const styles = theme => ({
  mainContainer: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center'
  },
  graph: {
    display: 'flex',
    justifyContent: 'center'
  },
  main: {
    width: '100%'
  },
  toggle: {
    paddingBottom: theme.spacing.unit * 3
  }
});

export default withStyles(styles)(Main);
