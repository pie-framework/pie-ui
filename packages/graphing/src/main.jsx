import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import uniq from 'lodash/uniq';
import {
  GraphContainer as Graph,
  tools as graphingTools
} from '@pie-lib/graphing';
import { Collapsible } from '@pie-lib/render-ui';
export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object,
    model: PropTypes.object.isRequired,
    marks: PropTypes.arrayOf(
      PropTypes.shape({ type: PropTypes.string.isRequired })
    ),
    onAnswersChange: PropTypes.func
  };

  render() {
    const { model, marks } = this.props;
    const tools = [];
    const marksTypes = marks ? marks.reduce((acc, m) => ([ ...acc, m.type]), []) : [];
    const backgroundMarksTypes = model.backgroundMarks ? model.backgroundMarks.reduce((acc, m) => ([ ...acc, m.type]), []) : [];
    const markTypes = uniq([ ...marksTypes, ...backgroundMarksTypes, ...(model.toolbarTools || []) ]);

    markTypes.forEach((graphingToolKey) => {
      if (typeof graphingTools[graphingToolKey] === 'function') {
        const tool = graphingTools[graphingToolKey]();
        tool.toolbar = !!(model.toolbarTools && model.toolbarTools.find(tT => tT === tool.type));
        tools.push(tool)
      }
    });

    const defaultAndCurrent = tools && tools.find(t => t.toolbar);
    return (
      <div>
        {
          model.teacherInstructions && (
            <Collapsible
              labels={{ hidden: 'Show Teacher Instructions', visible: 'Hide Teacher Instructions' }}
            >
              <div dangerouslySetInnerHTML={{ __html: model.teacherInstructions }}/>
            </Collapsible>
          )
        }
        <br />
        <Graph
          size={model.size}
          domain={model.domain}
          range={model.range}
          title={model.title}
          labels={model.labels}
          marks={marks}
          backgroundMarks={model.backgroundMarks}
          onChangeMarks={this.props.onAnswersChange}
          tools={tools}
          currentTool={defaultAndCurrent}
          defaultTool={defaultAndCurrent}
        />
        {
          model.rationale && (
            <Collapsible labels={{ hidden: 'Show Rationale', visible: 'Hide Rationale' }}>
              <div dangerouslySetInnerHTML={{ __html: model.rationale }}/>
            </Collapsible>
          )
        }
      </div>
    );
  }
}

const styles = () => ({});

export default withStyles(styles)(Main);
