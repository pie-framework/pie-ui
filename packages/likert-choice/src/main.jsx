import React from 'react';
import Radio from 'material-ui/Radio';
import Typography from 'material-ui/typography';
import {withStyles} from 'material-ui/styles';

const styles = {
    radioButton: {
        marginBottom: 16,
    },
    container: {
        verticalAlign: 'middle'
    },
    row: {
        display: 'inline-block',
        textAlign: 'center',
        paddingRight: '50px',
        paddingLeft: '50px'
    },
    textRow: {
        display: 'block'
    },
    identifier: {
        display: 'inline-flex',
        paddingRight: '1em',
        fontSize: '1em',
        fontWeight: 'bold'
    },
    label: {
        display: 'inline-flex',
    }
};

class Main extends React.Component {

    constructor(props){
        super(props);
        this.state= {
            radioValue: ''
        }
    }

    label(nameKey, myArray){
        for (var i=0; i < myArray.length; i++) {
            if (myArray[i].lang === nameKey) {
                return <div className={this.props.classes.label} dangerouslySetInnerHTML={{__html: myArray[i].label}}/>;
            }
        }
    }

    indexToSymbol(index) {
        return (this.props.model.keyMode === 'numbers'
                ? index + 1
                : String.fromCharCode(97 + index).toUpperCase()
        ).toString();
    }

    toggleRadio(e){
        this.props.onChoiceChanged(e);
        this.setState({radioValue: e.target.value});
    }


    render(){
        const {classes, model} = this.props;
        const {radioValue} = this.state;
        return (
            <div>
                <div dangerouslySetInnerHTML={{__html: model.prompt}}/>
                <div className={classes.container}>
                    {model.choices.map((k,index) => {
                        return(
                            <div key={index} className={classes.row}>
                                <Radio name='likert' onClick={(e) => this.toggleRadio(e)} checked={radioValue === k.value && k.value} value={k.value}/>
                                <div className={classes.textRow}>
                                    <Typography className={classes.identifier}>{this.indexToSymbol(index)}.</Typography>
                                    {this.label(model.activeLang, k.label)}
                                </div>
                            </div>
                        )
                    })}
                </div>
            </div>
        )
    }
}

export default withStyles(styles)(Main);