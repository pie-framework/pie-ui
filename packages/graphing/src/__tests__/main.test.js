import * as React from 'react';
import { Main } from '../main';
import { shallow } from 'enzyme/build';

jest.mock('lodash/uniq', () => {
  return () => ([]);
});

describe('Main', () => {
  const defaultProps = {
    model: {
      backgroundMarks: [],
      correctMarks: [],
    },
    onSessionChange: jest.fn(),
    onAnswersChange: jest.fn(),
    session: {}
  };

  describe('render', () => {
    let w;

    beforeEach(() => {
      w = props => shallow(<Main { ...props } />);
    });

    it('snapshot', () => {
      expect(w(defaultProps)).toMatchSnapshot();
    });
  });
});
