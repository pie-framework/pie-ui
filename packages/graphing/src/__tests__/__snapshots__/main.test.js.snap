// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main render snapshot 1`] = `
<div>
  <br />
  <Root
    backgroundMarks={Array []}
    onChangeMarks={[MockFunction]}
    tools={Array []}
  />
</div>
`;
